### PR TITLE
Get form notifications on a form

### PR DESCRIPTION
**Pull request by mVermaat**
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description
I'm my SpecFlow based test framework I got a functionality that retrieves form notifications. I think it could be a great addition to EasyRepro.

### Issues addressed
Some requirements add form notifications, you can automatically verify that they appear at the expected times. 

Secondary I also use it for logging when an error appears somewhere. Sometimes these notifications offer a good explaination what is going on. In example when your save fails due to a required field not being filled in. You will see that in the form notifications. Also some ribbon commands present a form notification instead of a dialog.

### All submissions:

- [X] My code follows the code style of this project.
- [X] Do existing samples that are effected by this change still run?
- [ ] I have added samples for new functionality. 
- [X] I raise detailed error messages when possible.
- [X] My code does not rely on labels that have the option to be hidden.

I didn't include any samples that I have to test this in my own framework, because they rely on customizations done on the environment that triggers these notifications. If you have some good ideas for a sample, I'm happy to add them to this pull request.

### Which browsers was this tested on?

- [X] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

**Comment by AngelRodriguez8008** [comment]
This can be a test:
1. Navigate to 'Leads'
2. Click 'New'
3. Click 'Save'
4. Assert Form Notification: 'Topic: Required fields must be filled in' 
image

5. (optional not yet supported) Assert Notification in 'subject' field: 'Required fields must be filled in' (it is required 

image

Note: 'subject' is required OOTB

File: Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs]: I prefer StringComparison.OrdinalIgnoreCase

Comment (AngelRodriguez8008) [reply Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs]: You don't really need this method. Move the logic to the class FormNotification, See previous comment

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: I thing you can move this if - else Logic to the FormNotification class. You don't need HasClass
Ex:
```csharp
string classes = icon.GetAttribute("class")
notification.SetTypeFromClass(classes);
```
In class FormNotification:
```csharp
        // I guest this list will grow, a dictionary is easier to read & maintain
        // note: keys are in lowercase
        Dictionary<string, FormNotificationType> MappingClassToTypes = new  Dictionary<string, FormNotificationType>{
            {"markaslost-symbol", FormNotificationType.Error},
            {"warning-symbol", FormNotificationType.Warning},
            {"informationicon-symbol", FormNotificationType.Information}
        };
        public void SetTypeFromClass(string classes) {
            string[] classesArr = classes.ToLower().Split(' '); 
            string classFound = classesArr.FirstOrDefault(c => MappingClassToTypes.ContainsKey(c));

            if(classFound == null)
                throw new InvalidOperationException($"Unknown notification type. Current class: {classes}");
            
            Type =  MappingClassToTypes[classFound];
        }
```

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: Replace TryFindElement with WaitUntilVisible (2 secs) just for the case that the notification requiere some time to fade-out.
```csharp
    var notificationBar = driver.WaitUntilVisible(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationBar]), 2.Secounds());
    if(notificationBar == null)
        return notifications;
```

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:3298]: Add a boolean variable, it is more readeable

**Review by AngelRodriguez8008 — CHANGES_REQUESTED** [review]
Great Job! I love your Idea of EasyRepro + SpecFlow

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (mVermaat) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: Incorporated

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (mVermaat) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs]: Never had any issues with that, however it makes sense. Better safe than sorry.

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs
Comment (mVermaat) [reply Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:3298]: Also added some additional comments to make it more clear.

**Review by AngelRodriguez8008 — DISMISSED** [review]
Great Job!

**Review by AngelRodriguez8008 — DISMISSED** [review]

**Review by TYLEROL — CHANGES_REQUESTED** [review]
Looks good - but requires a merge conflict fix before approval.

**Review by TYLEROL — APPROVED** [review]
Approving, merge conflicts resolved.

**Pull request merged**
